### PR TITLE
release: release notes and bumps for v18.1.0 release

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -51,146 +55,215 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
-- Added `hwm` in `getzmqnotifications`
-- `createwallet` can create blank, encrypted or watch-only wallets now.
-- Added `private_keys_enabled` in `getwalletinfo`
-- Added `solvable`, `desc`, `ischange` and `hdmasterfingerprint` in `getaddressinfo`
-- Added `desc` in `listunspent`
-
-Dash-specific changes in existing RPCs:
-- Added `quorumIndex` in `quorum getinfo` and `quorum memberof`
+### The new RPCs are:
+- The RPC `getrpcinfo` returns runtime details of the RPC server. At the moment
```
Comment (thephez) [inline doc/release-notes.md:174]: `getrpcinfo` is not a new RPC. It was in v18.0. 
But, `dumptxoutset`, `generatetodescriptor`, `generateblock`, `mockscheduler` (hidden), and `sporkupdate` are all new.

File: doc/release-notes.md [hunk 2]
```diff
@@ -51,146 +55,215 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
+* The wallet's `generate` RPC method has been deprecated and will be fully
+removed in v0.19. `generate` is only used for testing. The RPC call reaches across multiple
+subsystems (wallet and mining), so is deprecated to simplify the wallet-node
+interface. Projects that are using `generate` for testing purposes should
+transition to using the `generatetoaddress` call, which does not require or use
+the wallet component. Calling `generatetoaddress` with an address returned by
+`getnewaddress` gives the same functionality as the old `generate` method. 
+To continue using `generate` in v0.18, restart bitcoind with the
+`-deprecatedrpc=generate` configuration.
+- The wallet's `generate` RPC method was deprecated in v0.18 and has now
```
Comment (thephez) [inline doc/release-notes.md:188]: This and following line are duplicates (basically) of the previous paragraph

File: doc/release-notes.md [hunk 3]
```diff
@@ -51,146 +55,215 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
+    address's label will be changed to "final".
+  - You use the default settings to import a private key for an address that
+    was not previously in the wallet.  Its addresses will receive the default
+    empty label ("").
+- The `createwallet`, `loadwallet`, and `unloadwallet` RPCs now accept
+  `load_on_startup` options to modify the settings list.
+- Several RPCs have been updated to include an "avoid_reuse" flag, used to control
+  whether already used addresses should be left out or included in the operation.
+  These include:
+  - createwallet
```
Comment (thephez) [inline doc/release-notes.md:254]:
```suggestion
  - `createwallet`
```

File: doc/release-notes.md [hunk 4]
```diff
@@ -51,146 +55,215 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
+  - getbalance
```
Comment (thephez) [inline doc/release-notes.md:255]:
```suggestion
  - `getbalance`
```

File: doc/release-notes.md [hunk 5]
```diff
@@ -51,146 +55,215 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
+  - sendtoaddress
```
Comment (thephez) [inline doc/release-notes.md:256]:
```suggestion
  - `sendtoaddress`
```

**Review by thephez — CHANGES_REQUESTED** [review]
Some suggestions to consider. Also, it appears that some of the RPC changes actually were part of v18.0 :thinking:

File: doc/release-notes.md [hunk 6]
```diff
@@ -51,146 +55,215 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
+- `getnodeaddresses` returns peer addresses known to this node. It may be used to connect to nodes over TCP without using the DNS seeds.
+- Descriptors with key origin information imported through `importmulti` will have their key origin information stored in the wallet for use with creating PSBTs.
+- If `bip32derivs` of both `walletprocesspsbt` and `walletcreatefundedpsbt` is set to true but the key metadata for a public key has not been updated yet, then that key will have a derivation path as if it were just an independent key (i.e. no derivation path and its master fingerprint is itself)
+- The `getblockstats` RPC is faster for fee calculation by using BlockUndo data. Also, `-txindex` is no longer required and `getblockstats` works for all non-pruned blocks.
+- The `unloadwallet` RPC is now synchronous, meaning that it blocks until the wallet is fully unloaded.
+- `createwallet` can now create encrypted wallets if a non-empty passphrase is specified.
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply doc/release-notes.md:207]: I think so 🤷

File: doc/release-notes.md [hunk 7]
```diff
@@ -51,249 +55,251 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
+interface. Projects that are using `generate` for testing purposes should
+transition to using the `generatetoaddress` call, which does not require or use
+the wallet component. Calling `generatetoaddress` with an address returned by
+`getnewaddress` gives the same functionality as the old `generate` method.
+
+### Changes in existing RPCs introduced through bitcoin backports:
+* The `gettxoutsetinfo` RPC call now accept one optional argument (`hash_type`) that defines the algorithm used for calcluating the UTXO set hash, it will default to `hash_serialized_2` unless explicitly specified otherwise. `hash_type` will influence the key that is used to refer to refer to the UTXO set hash.
+* The `generatedescriptor` RPC call has been introduced to allow mining a set number of blocks, defined by argument `num_blocks`, with the coinbase destination set to a descriptor, defined by the `descriptor` argument. The optional `maxtries` argument can be used to limit iterations.
+- `getnodeaddresses` returns peer addresses known to this node. It may be used to connect to nodes over TCP without using the DNS seeds.
```
Comment (thephez) [inline doc/release-notes.md:196]: This also was added in 18.0. Was there a change?

Comment (PastaPastaPasta) [reply doc/release-notes.md:196]: Not sure, it's very likely we just didn't remove the release-note file

File: doc/release-notes.md [hunk 8]
```diff
@@ -51,249 +55,250 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
+- In rpc `upgradetohd` new parameter `rescan` was added which allows users to skip or force blockchain rescan. This params defaults to `false` when `mnemonic` parameter is empty and `true` otherwise.
 
 Please check `help <command>` for more detailed information on specific RPCs.
 
 Command-line options
 --------------------
+
 Most changes here were introduced through Bitcoin backports.
 
 New cmd-line options:
```
Comment (thephez) [inline doc/release-notes.md:258]: If there are no new options, this line should either be removed or there should be a message on the next line indicating there are no new one.

File: doc/release-notes.md [hunk 9]
```diff
@@ -51,249 +55,250 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
-- `keepassport`
-- `keepasskey`
-- `keepassid`
-- `keepassname`
-- `salvagewallet`
+
+-The `-zapwallettxes` startup option has been removed and its functionality removed from the wallet.
+  This option was originally intended to allow for the fee bumping of transactions that did not
+  signal RBF. This functionality has been superseded with the abandon transaction feature.
+- RPC Whitelist system. It can give certain RPC users permissions to only some RPC calls.
```
Comment (thephez) [inline doc/release-notes.md:265]: Is this new or an update?

File: doc/release-notes.md [hunk 10]
```diff
@@ -51,249 +55,250 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
 
 Removed cmd-line options:
-- `keepass`
-- `keepassport`
-- `keepasskey`
-- `keepassid`
-- `keepassname`
-- `salvagewallet`
+
+-The `-zapwallettxes` startup option has been removed and its functionality removed from the wallet.
```
Comment (thephez) [inline doc/release-notes.md:262]: Missing space messing up rendering
```suggestion
- The `-zapwallettxes` startup option has been removed and its functionality removed from the wallet.
```

**Review by thephez — CHANGES_REQUESTED** [review]

File: doc/release-notes.md (hunk 9 above)
Comment (PastaPastaPasta) [reply doc/release-notes.md:265]: I think new

File: doc/release-notes.md [hunk 11]
```diff
@@ -48,33 +55,249 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
 =======
 
 Thanks to everyone who directly contributed to this release:
 
-- kittywhiskers
+- Kittywhiskers Van Gogh
+- Konstantin Akimov
+- Munkybooty
+- Nathan Marley
+- Odysseas Gabrielides
```
Comment (UdjinM6) [inline doc/release-notes.md:293]:
```suggestion
- Odysseas Gabrielides
- Oleg Girko
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: doc/release-notes.md [hunk 12]
```diff
@@ -51,249 +55,249 @@ masternodes. This results in PoSe punishment/banning for outdated masternodes,
...
+- A new `setwalletflag` RPC sets/unsets flags for an existing wallet.
+- The `spork` RPC call will no longer offer both get (labelled as "basic mode") and set (labelled as "advanced mode") functionality. `spork` will now only offer "basic" functionality. "Advanced" functionality is now exposed through the `sporkupdate` RPC call.
+- The `generateblock` RPC call will mine an array of ordered transactions, defined by hex array `transactions` that can contain either transaction IDs or a hex-encoded serialized raw transaction and set the coinbase destination defined by the `address/descriptor` argument.
+- The `mockscheduler` is a debug RPC call that allows forwarding the scheduler by `delta_time`. This RPC call is hidden and will only be functional on mockable chains (i.e. primarily regtest). `delta_time` must be between 0 - 3600.
+- `dumptxoutset` see bitcoin#16899 and the help text. This RPC is used for UTXO snapshot creation which is a part of the assume utxo project
+- `generatetodescriptor` Mine blocks immediately to a specified descriptor. see bitcoin#16943
+
+### The removed RPCs are:
+- The wallet's `generate` RPC method has been removed. `generate` is only used for testing.
```
Comment (UdjinM6) [inline doc/release-notes.md:186]:
```suggestion
- The wallet's `generate` RPC method has been removed. `generate` is only used for testing.
```

**Review by thephez — DISMISSED** [review]
LGTM :+1:

**Comment by UdjinM6** [comment]
2 small issues, LGTM otherwise

**Comment by PastaPastaPasta** [comment]
Resolved both

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by thephez — APPROVED** [review]

**Pull request merged**